### PR TITLE
Introduce source_slug_mappings table

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary
In a previous PR (#1435) we changed `coinmarketcap_id` column to `slug`, which should be read as "santiment_id". So, the slug is no longer equal to the coinmarketcap_id. This will allow us to add projects that are not on CMC and to easily handle when a project on CMC gets changed.

So, we're introducing a mapping between a santiment project's slug and the slug of the same project on other sources. 

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase/model/project/slug_source_mapping.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  import Ecto.Changeset
+
+  alias Sanbase.Repo
+  alias Sanbase.Model.Project
+
+  @table "slug_source_mappings"
+
+  schema @table do
+    field(:source, :string)
+    field(:source_slug, :string)
```
Comment (tspenov) [inline lib/sanbase/model/project/slug_source_mapping.ex]: only `slug` :?

File: lib/sanbase/model/project/slug_source_mapping.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def changeset(%__MODULE__{} = ssm, attrs \\ %{}) do
+    ssm
+    |> cast(attrs, [:source, :source_slug, :project_id])
+    |> validate_required([:source, :source_slug, :project_id])
+    |> unique_constraint(:source_slug, name: :source_slug_unique_combination)
+  end
+
+  def get_source_slug(%Project{id: project_id}, source) do
```
Comment (tspenov) [inline lib/sanbase/model/project/slug_source_mapping.ex]: not used

File: lib/sanbase/web.ex [hunk 3]
```diff
@@ -19,7 +19,7 @@ defmodule Sanbase.Application.Web do
       Sanbase.TimescaleRepo,
 
       # Start the Clickhouse Repo
-      start_in({Sanbase.ClickhouseRepo, []}, [:dev, :prod]),
+      start_in({Sanbase.ClickhouseRepo, []}, [:prod]),
```
Comment (tspenov) [inline lib/sanbase/web.ex]: why removed ?

File: mix.exs [hunk 4]
```diff
@@ -84,7 +84,7 @@ defmodule Sanbase.Mixfile do
       {:observer_cli, "~> 1.3"},
       {:earmark, "~> 1.2"},
       {:ecto_enum, "~> 1.3"},
-      {:ex_machina, "~> 2.2", only: :test},
+      {:ex_machina, "~> 2.2", only: [:dev, :test]},
```
Comment (tspenov) [inline mix.exs]: do we need in :dev ?

File: priv/repo/migrations/20190903145409_add_slug_source_map_table.exs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+defmodule Sanbase.Repo.Migrations.AddSlugSourceMapTable do
+  use Ecto.Migration
+
+  @table "slug_source_mappings"
+  def change do
+    create table(@table) do
+      add(:source, :string)
+      add(:source_slug, :string)
```
Comment (tspenov) [inline priv/repo/migrations/20190903145409_add_slug_source_map_table.exs]: shouldn't we have non_null for both ?

File: lib/sanbase/web.ex (hunk 3 above)
Comment (IvanIvanoff) [reply lib/sanbase/web.ex]: I'll add it back, it's annoying while working locally without VPN on

File: lib/sanbase/model/project/slug_source_mapping.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/model/project/slug_source_mapping.ex]: I'll point the `coinmarketcap_id/1` from the project module to use it

File: mix.exs (hunk 4 above)
Comment (IvanIvanoff) [reply mix.exs]: It's much easier to test things locally with it rather than writing the changeset + repo inserts by hand

Comment (tspenov) [reply mix.exs]: then lets allow in dev 👍

File: lib/sanbase/model/project/slug_source_mapping.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase/model/project/slug_source_mapping.ex]: I don't like `source_slug` but also just `slug` could introduce some confusion :?

Comment (IvanIvanoff) [reply lib/sanbase/model/project/slug_source_mapping.ex]: After some thought, I renamed it to `slug` and changed the table name from `slug_source..` to `source_slug...`

**Pull request merged**